### ISSUE TITLE
Remove version from internal webserver

### ISSUE THREAD

**Issue by brdewildt** [PowerDNS/pdns#4979]
- Program: Authoritative, Recursor, dnsdist
 - Issue type: Feature request

### Short description

Please create an option to remove the versioning from the internal webservers.
Currently: 
  PowerDNS/0.0.999ge6e7dd8
Should read:
  PowerDNS

Just stripping of the version is enough to not be bothered again for security audits.

### Usecase

We are trying to harden all our software, removing the version headers is part of this.

### Description

Able to remove the version of the internal powerdns Webserver (or even modify the version display)

**Comment by rgacogne** [comment]
`dnsdist` is not affected since it doesn't send a `Server` header, but we should probably add an option to disable the version number in auth and rec.

**Comment by zeha** [comment]
If we do that, just drop everything except the major version from the string and be done with it?

**Comment by zeha** [comment]
(= don't have an option)

**Comment by mind04** [comment]
use the 'version-string' setting in the webserver

**Comment by zeha** [comment]
> version-string

Which will be something like `Served by PowerDNS - https://www.powerdns.com/` ...

**Comment by brdewildt** [comment]
version-string will only modify the output of the DNS response to version,
Not the output of the webserver itself.

Our GSA engine is reporting:

```
Vulnerability Detection Result
The remote web server type is :
PowerDNS/0.0.999ge6e7dd8 
```

WhereAs the DNS Server Detection is showing:
```
 Vulnerability Detection Result
The remote DNS server banner is:
IS TEST DNS PLATFORM
```

**Comment by Habbie** [comment]
I think @mind04's comment was a suggestion to whomever patches this, not something that works today.

**Comment by omoerbeek** [comment]
Fixed by #11813
